proposals/NativeInt.md

```diff
@@ -0,0 +1,326 @@
+| `string` | `nint` | None | |
+| `void*` | `nint` | PointerToVoid | `conv.i` |
+| `sbyte` | `nint` | ImplicitNumeric | `conv.i` |
+| `byte` | `nint` | ImplicitNumeric | `conv.u` |
+| `short` | `nint` | ImplicitNumeric | `conv.i` |
+| `ushort` | `nint` | ImplicitNumeric | `conv.u` |
+| `int` | `nint` | ImplicitNumeric | `conv.i` |
+| `uint` | `nint` | ExplicitNumeric | `conv.u` / `conv.ovf.u` |
+| `long` | `nint` | ExplicitNumeric | `conv.i` / `conv.ovf.i` |
+| `ulong` | `nint` | ExplicitNumeric | `conv.i` / `conv.ovf.i` |
+| `char` | `nint` | ImplicitNumeric | `conv.i` |
+| `float` | `nint` | ExplicitNumeric | `conv.i` / `conv.ovf.i` |
+| `double` | `nint` | ExplicitNumeric | `conv.i` / `conv.ovf.i` |
+| `decimal` | `nint` | ExplicitNumeric | `long IntPtr.op_Explicit(decimal) IntPtr IntPtr.op_Explicit(long)` |
+| `IntPtr` | `nint` | Identity | |
+| `UIntPtr` | `nint` | None | |
+| `object` | `nuint` | Unboxing | `unbox` |
+| `string` | `nuint` | None | |
+| `void*` | `nuint` | PointerToVoid | `conv.u` |
+| `sbyte` | `nuint` | ExplicitNumeric | `conv.u` / `conv.ovf.u` |
+| `byte` | `nuint` | ImplicitNumeric | `conv.u` |
+| `short` | `nuint` | ExplicitNumeric | `conv.u` / `conv.ovf.u` |
+| `ushort` | `nuint` | ImplicitNumeric | `conv.u` |
+| `int` | `nuint` | ExplicitNumeric | `conv.u` / `conv.ovf.u` |
+| `uint` | `nuint` | ImplicitNumeric | `conv.u` |
+| `long` | `nuint` | ExplicitNumeric | `conv.u` / `conv.ovf.u` |
+| `ulong` | `nuint` | ExplicitNumeric | `conv.u` / `conv.ovf.u` |
+| `char` | `nuint` | ImplicitNumeric | `conv.u` |
+| `float` | `nuint` | ExplicitNumeric | `conv.u` / `conv.ovf.u` |
+| `double` | `nuint` | ExplicitNumeric | `conv.u` / `conv.ovf.u` |
+| `decimal` | `nuint` | ExplicitNumeric | `ulong UIntPtr.op_Explicit(decimal) UIntPtr UIntPtr.op_Explicit(ulong)` |
+| `IntPtr` | `nuint` | None | |
+| `UIntPtr` | `nuint` | Identity | |
+
+| Operand | Target | Conversion | IL |
+|:---:|:---:|:---:|:---:|
+| `nint` | `object` | Boxing | `box` |
+| `nint` | `string` | None | |
+| `nint` | `void*` | PointerToVoid | `conv.i` |
+| `nint` | `nuint` | ExplicitNumeric | `conv.u` / `conv.ovf.u` |
+| `nint` | `sbyte` | ExplicitNumeric | `conv.i1` / `conv.ovf.i1` |
+| `nint` | `byte` | ExplicitNumeric | `conv.u1` / `conv.ovf.u1` |
+| `nint` | `short` | ExplicitNumeric | `conv.i2` / `conv.ovf.i2` |
+| `nint` | `ushort` | ExplicitNumeric | `conv.u2` / `conv.ovf.u2` |
+| `nint` | `int` | ExplicitNumeric | `conv.i4` / `conv.ovf.i4` |
+| `nint` | `uint` | ExplicitNumeric | `conv.u4` / `conv.ovf.u4` |
+| `nint` | `long` | ImplicitNumeric | `conv.i8` / `conv.ovf.i8` |
+| `nint` | `ulong` | ExplicitNumeric | `conv.i8` / `conv.ovf.i8` |
+| `nint` | `char` | ExplicitNumeric | `conv.u2` / `conv.ovf.u2` |
+| `nint` | `float` | ImplicitNumeric | `conv.r4` |
+| `nint` | `double` | ImplicitNumeric | `conv.r8` |
+| `nint` | `decimal` | ExplicitNumeric | `long decimal.op_Explicit(decimal) IntPtr IntPtr.op_Explicit(long)` |
```

Comment:
>ExplicitNumeric [](start = 23, length = 15)

Since the conversion from `long` to `decimal` is an implicit conversion, this should be too.

Reply:
From a future-proofing perspective, I don't think that makes sense.  `long` and `decimal` both have fixed sizes, and a `long` can always be represented as a `decimal` without any loss in precision--that's a strong guarantee.  That isn't the case for `nint`/`nuint` -> `decimal`.  If .NET Core ever lands on a platform with 128-bit native ints, that decision is going to be hard to correct.